packages/flutter/lib/src/material/button.dart

```diff
@@ -451,35 +452,111 @@ class MaterialButton extends StatelessWidget {
+  final double minWidth;
+  final double minHeight;
 
   @override
   RenderObject createRenderObject(BuildContext context) {
-    return new _RenderButtonRedirectingHitDetection(constraints);
+    return new _RenderButtonPadding(minWidth, minHeight);
   }
 
   @override
-  void updateRenderObject(BuildContext context, covariant _RenderButtonRedirectingHitDetection renderObject) {
-    renderObject.additionalConstraints = constraints;
+  void updateRenderObject(BuildContext context, covariant _RenderButtonPadding renderObject) {
+    renderObject
+      ..minHeight = minHeight
+      ..minWidth = minWidth;
   }
 }
 
-class _RenderButtonRedirectingHitDetection extends RenderConstrainedBox {
-  _RenderButtonRedirectingHitDetection (BoxConstraints additionalConstraints) : super(additionalConstraints: additionalConstraints);
+class _RenderButtonPadding extends RenderShiftedBox {
+  _RenderButtonPadding(this._minWidth, this._minHeight, [RenderBox child]) : super(child) ;
+
+  double get minWidth => _minWidth;
+  double _minWidth;
+  set minWidth(double value) {
+    if (_minWidth == value)
+      return;
+    _minWidth = value;
+    markNeedsLayout();
+  }
+
+  double get minHeight => _minHeight;
+  double _minHeight;
+  set minHeight(double value) {
+    if (_minHeight == value)
+      return;
+    _minHeight = value;
+    markNeedsLayout();
+  }
+
+  @override
+  double computeMinIntrinsicWidth(double height) {
+    if (child != null)
+      return math.max(child.computeMinIntrinsicWidth(height), minWidth);
+    return 0.0;
+  }
+
+  @override
+  double computeMinIntrinsicHeight(double width) {
+    if (child != null)
+      return math.max(child.computeMaxIntrinsicHeight(width), minHeight);
+    return 0.0;
+  }
+
+  @override
+  double computeMaxIntrinsicWidth(double height) {
+    if (child != null)
+      return math.max(child.computeMinIntrinsicWidth(height), minWidth);
+    return 0.0;
+  }
+
+  @override
+  double computeMaxIntrinsicHeight(double width) {
+    if (child != null)
+      return math.max(child.computeMaxIntrinsicHeight(width), minHeight);
+    return 0.0;
+  }
+
+  @override
+  void performLayout() {
+    if (child != null) {
+      child.layout(constraints, parentUsesSize: true);
+      size = constraints.constrain(new Size(math.max(child.size.width, minWidth), math.max(child.size.height, minHeight)));
+      final BoxParentData childParentData = child.parentData;
+      childParentData.offset = Alignment.center.alongOffset(size - child.size);
+    } else {
+      size = Size.zero;
+    }
+  }
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    if (child != null) {
+      final BoxParentData childParentData = child.parentData;
+      context.paintChild(child, offset + childParentData.offset);
+    }
+  }
+
+  @override
+  void setupParentData(RenderObject child) {
+    child.parentData = new BoxParentData();
+  }
 
   @override
   bool hitTest(HitTestResult result, {Offset position}) {
```

Comment:
It seeems like this method should be defined in terms of super.hitTest():
```
return super.hitTest(result, position: position)) || child.hitTest(result, position: child.size.center(Offset.zero));
```
That's assuming that this method is never called unless `size.contains(position)`.

Reply:
Done